meilisearch-http/src/index/mod.rs

```diff
@@ -83,6 +83,20 @@ impl Index {
             })
             .transpose()?
             .unwrap_or_else(BTreeSet::new);
+
+        // in milli each word in the synonyms map were split on their separator. Since we lost
+        // this information we are going to put space between words.
```

Comment:
```suggestion
        // Since milli stores the synonyms split by whitespaces, we need to reconstruct them before we return them to the user. This is a best effort approach.
```